test/integration/running/test_running.py

```diff
@@ -139,3 +142,50 @@ def test_logrotate(tt_cmd, tmpdir):
     # Check that the process was terminated correctly.
     instance_process_rc = instance_process.wait(1)
     assert instance_process_rc == 0
+
+
+def test_running_reread_config(tt_cmd, tmpdir):
+    # Copy the test application to the "run" directory.
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_app", "test_app_restart.lua")
+    instPath = shutil.copy(test_app_path, tmpdir)
+    random_name = 'test_app_'
+    random_name += str(randint(0,9))
+    randPath = os.path.join(tmpdir, random_name +'.lua')
+    configPath = os.path.join(tmpdir, "tarantool.yaml")
+    os.rename(instPath, randPath)
+
+    # Create test config with restart_on_failure true.
+    with open(configPath, 'w') as f:
+        f.write('tt:\n  app:\n    restart_on_failure: true')
+
+    # Start an instance.
+    start_cmd = [tt_cmd, "start", random_name, "--cfg", configPath]
+    instance_process = subprocess.Popen(
+        start_cmd,
+        cwd=tmpdir,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        text=True
+    )
+    start_output = instance_process.stdout.readline()
+    assert re.search(r"Starting an instance", start_output)
+
+    # Check status.
+    status_cmd = [tt_cmd, "status", random_name]
+    status_rc, status_out = run_command_and_get_output(status_cmd, cwd=tmpdir)
+    assert status_rc == 0
+    assert re.search(r"RUNNING. PID: \d+.", status_out)
+
+    with open(configPath, 'w') as f:
+        f.write('tt:\n  app:\n    restart_on_failure: false')
+    time.sleep(5)
```

Comment:
As far as I understand it, you just wait for the instance to finish executing the code and then check to see if it has been restarted.
This is incorrect. I think the code should not be restarted if the instance was completed correctly. Sounds like a bug in watchdog, I will discuss this point with PMs.

Reply:
I think we should discuss with a voice how best to stop the instance here.

Reply:
disscused in a voice call